graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/BuiltinFunctions.java

```diff
@@ -286,6 +294,209 @@ public Object absObject(VirtualFrame frame, Object object,
+
+            for (int i = 0; i < sequenceStorage.length(); i++) {
+                if (nodeType == NodeType.ALL && !isTrueNode.execute(frame, internalArray[i])) {
+                    return false;
+                } else if (nodeType == NodeType.ANY && isTrueNode.execute(frame, internalArray[i])) {
+                    return true;
+                }
+            }
+
+            return nodeType == NodeType.ALL;
+        }
+
+        @Specialization
+        boolean doIntSequence(VirtualFrame frame,
+                        IntSequenceStorage sequenceStorage,
+                        NodeType nodeType) {
+            int[] internalArray = sequenceStorage.getInternalIntArray();
+
+            for (int i = 0; i < sequenceStorage.length(); i++) {
+                if (nodeType == NodeType.ALL && !isTrueNode.execute(frame, internalArray[i])) {
+                    return false;
+                } else if (nodeType == NodeType.ANY && isTrueNode.execute(frame, internalArray[i])) {
+                    return true;
+                }
+            }
+
+            return nodeType == NodeType.ALL;
+        }
+
+        @Specialization
+        boolean doGenericSequence(VirtualFrame frame,
+                        SequenceStorage sequenceStorage,
+                        NodeType nodeType,
+                        @Cached SequenceStorageNodes.LenNode lenNode) {
+            Object[] internalArray = sequenceStorage.getInternalArray();
+            for (int i = 0; i < lenNode.execute(sequenceStorage); i++) {
+                if (nodeType == NodeType.ALL && !isTrueNode.execute(frame, internalArray[i])) {
+                    return false;
+                } else if (nodeType == NodeType.ANY && isTrueNode.execute(frame, internalArray[i])) {
+                    return true;
+                }
+            }
+
+            return nodeType == NodeType.ALL;
+        }
+
+        @Specialization(limit = "3")
+        protected boolean doHashStorage(VirtualFrame frame,
+                        HashingStorage hashingStorage,
+                        NodeType nodeType,
+                        @CachedLibrary("hashingStorage") HashingStorageLibrary hlib) {
+            for (Object key : hlib.keys(hashingStorage)) {
+                if (nodeType == NodeType.ALL) {
+                    if (!isTrueNode.execute(frame, key)) {
+                        return false;
+                    }
+                } else if (nodeType == NodeType.ANY && isTrueNode.execute(frame, key)) {
+                    return true;
+                }
+            }
+
+            return nodeType == NodeType.ALL;
+        }
+    }
+
+    @Builtin(name = ALL, minNumOfPositionalArgs = 1)
+    @GenerateNodeFactory
+    public abstract static class AllNode extends PythonUnaryBuiltinNode {
+
+        @Specialization(guards = "cannotBeOverridden(object, getClassNode)", limit = "1")
+        boolean doList(VirtualFrame frame,
```

Comment:
The specialization methods here and in `AnyNode` could be `static`.

Reply:
Corrected in aa7cde4, but out of curiosity, is there a performance benefit to these methods being static? Or is it a code style/quality decision

Reply:
There can be a small performance benefit in the interpreter for startup. In Java, invoking static methods can be a bit faster because there is no overriding, it doesn't have to lookup which method to call (there are two separate JVM instructions - `invokestatic` and `invokevirtual`). After it gets compiled by graal, there is no difference (the node types are constant and the method will get inlined, so there's no method call in the compiled code at all).